### ISSUE TITLE
Fast way to load multiple models from files?

### ISSUE THREAD

**Issue by yotam180** [tflearn/tflearn#974]
I'm building a system for classifying images into multiple categories, and approaching the problem using multiple networks (all my networks have the same structure, and output `[0,1]` or `[1,0]` (Yes or No).

There's (for the sake of the example) a network for dogs, another network for cats, the third network is for cars, and so and so...

**My first question**: Is this kind of approach the popular one, or is there a different (and hopefully better) approach for classifying a LOT of different subjects?

If I'm on the right path, I wondered if there a fast way to load the models from their files using TFLearn.

Currently, I have a function to define my model structure:

    def define_network_model(category_id):
        """
        Defines a network structure
        """`
    
        # Defining image augmentation
        # We want to create a stilted wider dataset
        aug = ImageAugmentation()
        aug.add_random_flip_leftright()
        aug.add_random_blur(sigma_max=3.)
        aug.add_random_rotation(max_angle=20.)
        
        T.reset_default_graph()
    
        # Input layer
        net = input_data(shape=[None, IMG_SIZE, IMG_SIZE, 1], name="input", \
            data_augmentation=aug)
    
        # 2D Convolution layer
        net = conv_2d(net, 32, 5, activation='relu')
        net = max_pool_2d(net, 3)
    
        .... More layers ....
    
        # Output layer
        net = fully_connected(net, 2, activation='softmax')
    
        # Regression layer
        net = regression(net, optimizer='adam', learning_rate=LEARNING_RATE, \
            loss='categorical_crossentropy', name='targets')
        
        # Creating a model
        model = tflearn.DNN(net, tensorboard_dir="tensorboard", \
            best_checkpoint_path=os.path.join(CHECKPOINTS_DIR, category_id), \
            best_val_accuracy=0.7)
    
        return model

And when I want to load the models, I have a categories object I'm iterating through and loading all the categories:

    models = {}
    
    for id, cat in categories.items():
        m = define_network_model(id) # Creating the structure
        m.load(os.path.join(MODELS_DIR, id)) # Loading the file
        models[id] = m # Appending to the models list for later evaluation

Defining the network every iteration takes **a LOT** of time (>2sec for every iteration).

**My second question**: Am I missing something here? Can the structure be cloned from a model to another in a faster way, or am I just on the wrong path?

**Comment by shreyas-7** [comment]
I have the same problem. Can anyone comment?
